src/main/java/net/fabricmc/loom/api/LoomGradleExtensionAPI.java

```diff
@@ -31,36 +31,88 @@
 
 import net.fabricmc.loom.api.decompilers.LoomDecompiler;
 import net.fabricmc.loom.configuration.ide.RunConfigSettings;
 import net.fabricmc.loom.configuration.processors.JarProcessor;
 import net.fabricmc.loom.configuration.providers.mappings.LayeredMappingSpecBuilder;
+import net.fabricmc.loom.util.DeprecationHelper;
 
 /**
  * This is the public api available exposed to build scripts.
  */
 public interface LoomGradleExtensionAPI {
-	File getAccessWidener();
+	@ApiStatus.Internal
+	DeprecationHelper getDeprecationHelper();
 
-	void setAccessWidener(File file);
+	RegularFileProperty getAccessWidenerPath();
 
-	void setShareCaches(boolean shareCaches);
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default File getAccessWidener() {
+		getDeprecationHelper().replaceWithInLoom0_11("accessWidener", "accessWidenerPath");
+		return getAccessWidenerPath().getAsFile().getOrNull();
+	}
+
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default void setAccessWidener(File file) {
+		getDeprecationHelper().replaceWithInLoom0_11("accessWidener", "accessWidenerPath");
+		getAccessWidenerPath().set(file);
+	}
+
+	Property<Boolean> getShareRemapCaches();
+
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default void setShareCaches(boolean shareCaches) {
+		getDeprecationHelper().replaceWithInLoom0_11("shareCaches", "shareRemapCaches");
+		getShareRemapCaches().set(shareCaches);
+	}
 
-	boolean isShareCaches();
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default boolean isShareCaches() {
+		getDeprecationHelper().replaceWithInLoom0_11("shareCaches", "shareRemapCaches");
+		return getShareRemapCaches().get();
+	}
 
 	default void shareCaches() {
-		setShareCaches(true);
+		getShareRemapCaches().set(true);
 	}
 
-	List<LoomDecompiler> getDecompilers();
+	ListProperty<LoomDecompiler> getGameDecompilers();
+
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default List<LoomDecompiler> getDecompilers() {
+		getDeprecationHelper().replaceWithInLoom0_11("decompilers", "gameDecompilers");
+		return getGameDecompilers().get();
+	}
 
-	void addDecompiler(LoomDecompiler decompiler);
+	default void addDecompiler(LoomDecompiler decompiler) {
+		getGameDecompilers().add(decompiler);
+	}
 
-	List<JarProcessor> getJarProcessors();
+	ListProperty<JarProcessor> getGameJarProcessors();
 
-	void addJarProcessor(JarProcessor processor);
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default List<JarProcessor> getJarProcessors() {
+		getDeprecationHelper().replaceWithInLoom0_11("jarProcessors", "gameJarProcessors");
+		return getGameJarProcessors().get();
+	}
+
+	@Deprecated(forRemoval = true)
+	@ApiStatus.ScheduledForRemoval(inVersion = "0.11")
+	default void addJarProcessor(JarProcessor processor) {
```

Comment:
The comment about utility methods also applies here (sorry, didn't notice this before) but it's really just a matter of opinion whether these are kept.